vllm build failure on IBM ppc64le

### Your current environment
IBM powerpc64le.
RHEL 9/ubi9
vllm: Built from source main branch.

### How you are installing vllm

```sh
docker build -t vllm:latest -f Dockerfile.ppc64le .
```
Error:
```

51.54     cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
51.54 
51.54 
51.54     Could not find openssl via pkg-config:
51.54 
51.54     pkg-config exited with status code 1
51.54     > PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --libs --cflags openssl
51.54 
51.54     The system library `openssl` required by crate `openssl-sys` was not found.
51.54     The file `openssl.pc` needs to be installed and the PKG_CONFIG_PATH environment variable must contain its parent directory.
51.54     The PKG_CONFIG_PATH environment variable is not set.
51.54 
51.54     HINT: if you have installed the library, try setting PKG_CONFIG_PATH to the directory containing `openssl.pc`.
51.55 
51.55 
51.55     cargo:warning=Could not find directory of OpenSSL installation, and this `-sys` crate cannot proceed without this knowledge. If OpenSSL is installed and this crate had trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the compilation process. See stderr section below for further information.
51.55 
51.55     --- stderr
51.55 
51.55 
51.55     Could not find directory of OpenSSL installation, and this `-sys` crate cannot
51.55     proceed without this knowledge. If OpenSSL is installed and this crate had
51.55     trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the
51.55     compilation process.
51.55 
51.55     Make sure you also have the development packages of openssl installed.
51.55     For example, `libssl-dev` on Ubuntu or `openssl-devel` on Fedora.
51.56 
51.56     If you're in a situation where you think the directory *should* be found
51.56     automatically, please open a bug at https://github.com/sfackler/rust-openssl
51.56     and include information about your system as well as this message.
51.56 
51.56     $HOST = powerpc64le-unknown-linux-gnu
51.56     $TARGET = powerpc64le-unknown-linux-gnu
51.56     openssl-sys = 0.9.104
51.56 
51.56 
51.56   warning: build failed, waiting for other jobs to finish...
63.11   error: `cargo rustc --lib --message-format=json-render-diagnostics --manifest-path Cargo.toml --release -v --features 'pyo3/extension-module python-bindings' --crate-type cdylib -- --crate-type=cdylib` failed with code 101
63.14   error: subprocess-exited-with-error
63.14   
63.14   × Building wheel for outlines_core (pyproject.toml) did not run successfully.
63.14   │ exit code: 1
63.14   ╰─> See above for output.

....
....

70.55   Created wheel for httptools: filename=httptools-0.6.4-cp310-cp310-linux_ppc64le.whl size=130467 sha256=84914d86dae1fee9efa3cd10af06f5594ee7cfe16fb189af7b0acb7522b26a95
70.55   Stored in directory: /root/.cache/pip/wheels/2a/2e/af/680e384ed0eb6f474586969a6552a57bd8d8608477364468b1
70.55 Successfully built httptools
70.55 Failed to build outlines_core
70.55 ERROR: Could not build wheels for outlines_core, which is required to install pyproject.toml-based projects
------
Dockerfile.ppc64le:22
--------------------
  21 |     # These packages will be in rocketce eventually
  22 | >>> RUN --mount=type=cache,target=/root/.cache/pip  \
  23 | >>>     pip install -v --prefer-binary --extra-index-url https://repo.fury.io/mgiessing \
  24 | >>>         'cmake>=3.26' ninja packaging 'setuptools-scm>=8' wheel jinja2 \
  25 | >>>         torch==2.4.1 \
  26 | >>>         -r requirements-cpu.txt \
  27 | >>>         xformers uvloop==0.20.0
  28 |     
--------------------
ERROR: failed to solve: process "/usr/local/bin/_dockerfile_shell.sh pip install -v --prefer-binary --extra-index-url https://repo.fury.io/mgiessing         'cmake>=3.26' ninja packaging 'setuptools-scm>=8' wheel jinja2         torch==2.4.1         -r requirements-cpu.txt         xformers uvloop==0.20.0" did not complete successfully: exit code: 1
```

Even after installing `libssl-dev`, outlines_core fails to locate the libraries with following error -
```
136.0     = note: LC_ALL="C" PATH="/opt/conda/lib/rustlib/powerpc64le-unknown-linux-gnu/bin:/tmp/pip-build-env-fw2vfbb3/overlay/bin:/tmp/pip-build-env-fw2vfbb3/normal/bin:/opt/conda/bin:/opt/conda/condabin:/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/conda/bin/" VSLANG="1033" "/opt/conda/bin/powerpc64le-conda-linux-gnu-cc" "-Wl,--version-script=/tmp/rustchY3nxX/list" "-Wl,--no-undefined-version" "-m64" "/tmp/rustchY3nxX/symbols.o" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/deps/outlines_core.outlines_core.4d42cff6ba60f38d-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/deps" "-L" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/build/ring-e51f19535f8bb199/out" "-L" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/build/esaxx-rs-ff5db49b271e086d/out" "-L" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/build/onig_sys-10a029a8b8a442ca/out" "-L" "/opt/conda/lib/rustlib/powerpc64le-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/rustchY3nxX/libesaxx_rs-7f74a07230d29c29.rlib" "/tmp/rustchY3nxX/libonig_sys-b5be662fa0ab51dc.rlib" "/tmp/rustchY3nxX/libring-8a331d784123b0f3.rlib" "/opt/conda/lib/rustlib/powerpc64le-unknown-linux-gnu/lib/libcompiler_builtins-48f1365f3e71a657.rlib" "-Wl,-Bdynamic" "-lstdc++" "-lssl" "-lcrypto" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/opt/conda/lib/rustlib/powerpc64le-unknown-linux-gnu/lib" "-o" "/tmp/pip-install-q5sk0wnm/outlines-core_270e564fa5b94bfdac308631b28baba9/target/release/deps/liboutlines_core.so" "-Wl,--gc-sections" "-shared" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-Wl,--strip-all" "-nodefaultlibs"
136.0     = note: /opt/conda/bin/../lib/gcc/powerpc64le-conda-linux-gnu/11.2.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: cannot find -lssl: No such file or directory
136.0             /opt/conda/bin/../lib/gcc/powerpc64le-conda-linux-gnu/11.2.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: cannot find -lcrypto: No such file or directory
136.0             collect2: error: ld returned 1 exit status
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/Dockerfile.ppc64le
+++ b/Dockerfile.ppc64le
@@ -4,7 +4,7 @@ USER root
 
 ENV PATH="/usr/local/cargo/bin:$PATH:/opt/conda/bin/"
 
-RUN apt-get update -y && apt-get install -y git wget curl vim libnuma-dev libsndfile-dev libprotobuf-dev build-essential ffmpeg libsm6 libxext6 libgl1 
+RUN apt-get update -y && apt-get install -y git wget curl vim libnuma-dev libsndfile-dev libprotobuf-dev build-essential ffmpeg libsm6 libxext6 libgl1 libssl-dev 
 
 # Some packages in requirements-cpu are installed here
 # IBM provides optimized packages for ppc64le processors in the open-ce project for mamba
@@ -18,9 +18,8 @@ ARG GIT_REPO_CHECK=0
 RUN --mount=type=bind,source=.git,target=.git \
     if [ "$GIT_REPO_CHECK" != 0 ]; then bash tools/check_repo.sh; fi
 
-# These packages will be in rocketce eventually
 RUN --mount=type=cache,target=/root/.cache/pip  \
-    pip install -v --prefer-binary --extra-index-url https://repo.fury.io/mgiessing \
+    RUSTFLAGS='-L /opt/conda/lib' pip install -v --prefer-binary --extra-index-url https://repo.fury.io/mgiessing \
         'cmake>=3.26' ninja packaging 'setuptools-scm>=8' wheel jinja2 \
         torch==2.3.1 \
         -r requirements-cpu.txt \

